transport/tlscommon/tls_fips_test.go

```diff
@@ -29,6 +31,9 @@ import (
 
 // TestFIPSCertifacteAndKeys tests that encrypted private keys fail in FIPS mode
 func TestFIPSCertificateAndKeys(t *testing.T) {
+	if checkGODEBUG() {
+		t.Skip("GODEBUG=fips140=only detected, avoiding panics")
+	}
```

Comment:
Please add a comment as to why this test needs to be excluded for `fips140=only`. As these are the FIPS tests (not the nonfips ones) it requires a nuanced understanding of the various fips related testing strategies to understand why fips tests are skipped in this scenario.

Reply:
+1 to the above, I cannot fully decode why this test under `requirefips` tag requires this check 🙂

Reply:
added the comment, moved the check to a subtest; and increased password length so the other subtest passes